app/utils/modify-shared-ways.js

```diff
@@ -139,7 +139,10 @@ function mergeNode (sharedWays, sourceNode, destinationNode) {
     }
 
     if (!newWay.properties.mergedNodes.includes(destinationNode.properties.id)) {
```

Comment:
i think this will always return true because we need to test a property in an object in the array. something like this should work:

```js
if (!newWay.properties.mergedNodes.some(merge => merge.destinationNode === destinationNode.properties.id)) {
  // do the push
}
```

```suggestion
    if (!newWay.properties.mergedNodes.some(merge => merge.destinationNode === destinationNode.properties.id)) {
```

Reply:
Ah good catch @sethvincent. I think we don't need this check anymore because we have to push the source and destination everytime regardless. Assuming that ones a source node is merged, the user is likely not able to merge that node with another node.